src/root.jl

```diff
@@ -135,15 +135,7 @@ end
 
 
 function Base.getindex(f::ROOTFile, s::AbstractString)
```

Comment:
how are we gonna use this public API going forward?

Reply:
I'd start with your proposal (https://github.com/JuliaHEP/UnROOT.jl/issues/194#issuecomment-1312494186) to clarify

- file["tree"]
- file["tree"]["branch"]
- file["tree/branch"]
- LazyTree(file, "tree")

while I think that the first three should be equivalent. I guess it's fine to simply return whatever ROOT has sittig at that position and we can think of implementing `getindex()` for `TBranchElement` or so to make e.g. `file["tree/branch"][23]` work, similar to e.g. `HDF5.jl` or so.

What do you think?

Reply:
Maybe convert to "array" or "lazyarray" automatically if one reaches the end of a branching... not sure. I still think explicit calling to `array` or `LazyBranch` etc. is better

Reply:
Something like this (it can get prettier though, maybe condense the information and also print if it's "interpretable" or needs custom parser):

```
julia> f["E/Evt/trks"]
trks
├─ trks.fUniqueID
├─ trks.fBits
├─ trks.usr
├─ trks.usr_names
├─ trks.any
├─ trks.id
├─ trks.pos.x
├─ trks.pos.y
├─ trks.pos.z
├─ trks.dir.x
├─ trks.dir.y
├─ trks.dir.z
├─ trks.t
├─ trks.E
├─ trks.len
├─ trks.lik
├─ trks.type
├─ trks.rec_type
├─ trks.rec_stages
├─ trks.status
├─ trks.mother_id
├─ trks.fitinf
├─ trks.hit_ids
├─ trks.error_matrix
└─ trks.comment


julia> f["E/Evt/trks/trks.E"]
UnROOT.TBranchElement_10
  cursor: UnROOT.Cursor
  fName: String "trks.E"
  fTitle: String "E[trks_]"
  fFillColor: Int16 0
  fFillStyle: Int16 1001
  fCompress: Int32 1
  fBasketSize: Int32 5000000
  fEntryOffsetLen: Int32 384
  fWriteBasket: Int32 1
  fEntryNumber: Int64 96
  fIOFeatures: UnROOT.ROOT_3a3a_TIOFeatures
  fOffset: Int32 0
  fMaxBaskets: UInt32 0x0000000a
  fSplitLevel: Int32 0
  fEntries: Int64 96
  fFirstEntry: Int64 0
  fTotBytes: Int64 39902
  fZipBytes: Int64 1651
  fBranches: UnROOT.TObjArray
  fLeaves: UnROOT.TObjArray
  fBaskets: UnROOT.TObjArray
  fBasketBytes: Array{Int64}((10,)) [1651, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  fBasketEntry: Array{Int64}((10,)) [0, 96, 0, 0, 0, 0, 0, 0, 0, 0]
  fBasketSeek: Array{Int64}((10,)) [12855827, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  fFileName: String ""
  fClassName: String "Trk"
  fParentName: String "Trk"
  fClonesName: String ""
  fCheckSum: UInt32 0x620ae336
  fClassVersion: Int16 11
  fID: Int32 5
  fType: Int32 41
  fStreamerType: Int32 8
  fMaximum: Int32 0
  fBranchCount: Missing missing
  fBranchCount2: Missing missing

```

Reply:
yeah I'm fine with: "getindex() is for inspection of the ROOT object metadata, if you need physical data, you need `array` or `LazyTree`"

Reply:
OK, let me know if you think we can merge this and I'll update the docs accordingly before we push the button ;)